### ISSUE TITLE
A11y: Card Component Checkbox Should Have a Visible Label

### ISSUE THREAD

**Issue by r-watkins** [Esri/calcite-design-system#2426]
### Summary
When the card component is selectable the checkbox does not have a visible label.

![Screen Shot 2021-06-30 at 9 50 23 AM](https://user-images.githubusercontent.com/17993027/124170505-5aca8400-da5c-11eb-9e30-a82105299136.png)

### Actual Behavior

There is a Select/Deselect label that appears which I think serves screen readers and will provide context for most sighted users.

![Screen Shot 2021-07-01 at 11 07 26 AM](https://user-images.githubusercontent.com/17993027/124170850-bbf25780-da5c-11eb-9acc-70671e3706ca.png)

### Expected Behavior

It's my understanding that checkboxes should have a visible label that can be clicked to check/uncheck the checkbox to increase the clickable area and make it easier for those with a motor control impairment to perform the selection action. 

This is my typical card design which uses the card title as the label:

![Screen Shot 2021-07-01 at 11 01 10 AM](https://user-images.githubusercontent.com/17993027/124172704-fd840200-da5e-11eb-9c64-684a12d82d9e.png)

I wonder though if we could just make the entire card a clickable area to check/uncheck the checkbox and fulfill the same criteria of an increased selection area?

### Relevant Info 
I am referencing the card design here with the selectable option:
https://esri.github.io/calcite-components/?path=/story/components-card--simple

### Working W3C Example/Tutorial 
I am referencing this W3.org spec:
https://www.w3.org/WAI/WCAG21/Techniques/html/H44

**Comment by julio8a** [comment]
We just had an accessibility audit from a third-party company. Will check with them on the a11y requirements for something like this.

**Comment by macandcheese** [comment]
It seems like the "Actual Behavior" referenced in the OP no longer works - we should ensure that functionality is working.

**Comment by driskull** [comment]
@benelan @jcfranco I think this should go through design to position the label alongside the checkbox.

We don't want to just implement it without consulting the design team and updating figma/etc.

**Comment by jcfranco** [comment]
We did triage this together, but I'll defer to @macandcheese.

**Comment by macandcheese** [comment]
We decided during triage not to address the add a visible label portion of the request. However the “actual behavior” in the OP doesn’t work - we should at least have a native label show up as it used to.

**Comment by eriklharper** [comment]
> I wonder though if we could just make the entire card a clickable area to check/uncheck the checkbox and fulfill the same criteria of an increased selection area?

Just now seeing this fairly old issue, but the `tile-select` component has very similar UX to the card component's checkbox mode, and the entire "tile" is clickable and focusable (both functionally and visibly):

![image](https://user-images.githubusercontent.com/821864/160889483-2efb6fb2-f36d-43d6-9df6-665fd15c7a81.png)

Sounds like a good UX discussion to be had where we take into account consistency between clickable area between these two similar components.  I agree that the entire area should be clickable and visually focusable, and in the case of the card, the `title` should probably serve as the label?

Another related design discussion I've had recently with @asangma is moving the checkbox into the card itself in all cases so that the checkbox never appears on top of the image, which will better visually align the box with the text content for sighted users to make better sense out of the meaning of the checkbox.  I'm currently working on this via #3181 

![image](https://user-images.githubusercontent.com/821864/160892001-174d67df-eb42-41e2-a27c-5b673f58d376.png)

**Comment by macandcheese** [comment]
Part of the current distinction between the two components is the difference in click areas. Tiles serve a different purpose than cards, which can often include multiple interactive areas. 

Making the entire card clickable could be discussed, but IMO - that can often lead to misinterpretation of hitbox areas when apps add "subtitles as link to author page", footer actions, etc.

That said, some of the hover effects our card has conflict with the analogy of the component and the current hit areas - they insinuate the entire card is clickable when it isn't. This should be looked at!

**Comment by driskull** [comment]
Maybe we could bump up the design priority on this one? We have it in our current milestone but we can't do anything until we get the design decisions made. @jcfranco @yelenakreyndel

**Comment by yelenakreyndel** [comment]
@driskull thank you for addressing this. We now have officially a dedicated designer who handles component enhancements. That's @ashetland. Matt, based on your comment, it sounds like this should be a 1.0 enhancement? @jcfranco

**Comment by yelenakreyndel** [comment]
@driskull These are the prioritized enhancements Aaron is working on this year: https://confluencewikidev.esri.com/display/Calcite/Enhancement+Requests 
Should this issue be added too? @jcfranco

**Comment by driskull** [comment]
I'll defer to @jcfranco for the priority

**Comment by ashetland** [comment]
Thanks @yelenakreyndel for tagging me here. I'm going to add this to our Figma doc so it's documented there and ready to go when @jcfranco and the team set a priority.

**Comment by jcfranco** [comment]
Can we clarify what this issue entails now? The entire thread has multiple suggestions. This is what I'm seeing:

> We decided during triage not to address the add a visible label portion of the request. 

The visible label is no longer in scope.

> However the “actual behavior” in the OP doesn’t work - we should at least have a native label show up as it used to.

- [ ] Setting `title` on the checkbox to get the native tooltip. Do we need this if the aria-label is set accordingly?

> That said, some of the hover effects our card has conflict with the analogy of the component and the current hit areas - they insinuate the entire card is clickable when it isn't. This should be looked at!

- [ ] Avoiding click confusion from hover effects 😵‍💫 This would be good to tackle before 1.0.0.

Let me know if I missed anything.

**Comment by ashetland** [comment]
@jcfranco The only other suggestion if found was this:
> moving the checkbox into the card itself in all cases so that the checkbox never appears on top of the image
This is definitely something to consider from an accessibility standpoint as well. Without a visible label it just looks like a box over the image. 

I'd also like to add that maybe we should look at the need for a checkbox when the footer contains buttons. This seems like a very confusing state. The card is selectable and it has buttons that do something else?? I'm curious if any products use that state and if the users understand the difference in context.

**Comment by ashetland** [comment]
@jcfranco Sorry, looks like moving the checkbox has already been done. But the other issue might be worth discussing.

**Comment by macandcheese** [comment]
It is pretty common to have other actions in a card when selectable (for instance these from Online):
<img width="847" alt="Screen Shot 2022-07-28 at 12 31 15 PM" src="https://user-images.githubusercontent.com/4733155/181622986-6f8482e4-bc20-4ccc-859c-dc05ef412b1a.png">

Also shown, users often place links to a detail page, author page, etc., in the title / subtitle slots which have their own hitboxes. For "one big hitbox" - Tile / Tile Select for selection is the recommended component.

**Comment by ashetland** [comment]
Thanks Adam, that context helps a lot!

**Comment by geospatialem** [comment]
The checkbox selection/de-selection and context to the component surfaced in the a11y audit, where the priority is set as something we should aim to address before the v1 release. 

Where the user impact includes: 
> Screen reader users will be unable to determine that these controls can be checked or unchecked. They will also be unable to review which options they have chosen.

However, this issue seems a bit beyond what is needed to achieve accessibility for the component, and might be out of scope. 

Once the a11y audit issues are created, would recommend blocking this issue until we're able to address the issue from the audit. Where, once completed, we could re-evaluate the ask above to determine feasibility.

cc: @jcfranco @macandcheese

**Comment by geospatialem** [comment]
@SkyeSeitz + @ashetland: We'd like to explore an enhancement for a larger hit area on the `calcite-card`s selectable checkbox. 

WDYT on implementing for a better user experience and further support of a11y?

## Current experience
selectable `calcite-card` checkbox:

![selectablecard-hitarea](https://user-images.githubusercontent.com/5023024/195939377-dd51fc48-1c54-4dce-98fc-679f910334d4.gif)

**Comment by ashetland** [comment]
Here's a quick mock that increases the hit area to 30px. Basically using the existing margin. Thoughts?
<img width="305" alt="CleanShot 2022-10-18 at 12 10 21@2x" src="https://user-images.githubusercontent.com/108549080/196522561-19762544-c579-4216-a2ae-40e89eb7ee29.png">

**Comment by macandcheese** [comment]
We could also use `--calcite-checkbox-size` to increase the size of the component.

**Comment by ashetland** [comment]
Using `--calcite-checkbox-size` could be a good option as well, but I think we could have a better default.

There's a pretty big discrepancy between the hit area and the Checkbox in my above example. While this may work for certain button types, it feels too big for a checkbox.

Another option could be to use `scale="l"` for Checkbox, increasing its size to 16px. 
<img width="313" alt="CleanShot 2022-10-18 at 15 54 02@2x" src="https://user-images.githubusercontent.com/108549080/196560068-5ccbc3e5-bc4b-4e0d-a226-5ffeb7126fca.png">

If a larger hit area is desired, then we could then expand it to 24px without it being awkward since it's very close to the focus outline. This would bring it closer to how Carbon handles a checkbox without a label, but theirs is 20 x 24px.
<img width="313" alt="CleanShot 2022-10-18 at 15 53 52@2x" src="https://user-images.githubusercontent.com/108549080/196560567-b0c67b7a-7f7f-4674-a056-825b5983b7ff.png">
Carbon:
<img width="284" alt="CleanShot 2022-10-18 at 16 08 23@2x" src="https://user-images.githubusercontent.com/108549080/196561677-12387f50-da0e-4a32-a71d-4fb444432672.png">

**Comment by SkyeSeitz** [comment]
My vote is to both increase the scale of the checkbox to `scale="l"` and increase the hit area to at least 24px if not 30px since the typical recommended minimum target size is 44 pixels. 
<img width="332" alt="image" src="https://user-images.githubusercontent.com/29716057/198734971-7968fc80-6e03-4568-bc1e-622e5f6c915c.png">

I am not sure the extra pixels on the hit area hurt since the card design safely spaces out the checkbox with a good amount of breathing room. 
<img width="330" alt="image" src="https://user-images.githubusercontent.com/29716057/198735034-739407b3-351c-49ca-8739-c785729452ab.png">
Thoughts?

**Comment by jcfranco** [comment]
My vote is for @SkyeSeitz's vote. 🏆

**Comment by ashetland** [comment]
Based on this https://www.w3.org/TR/WCAG22/#target-size-minimum (Thanks @geospatialem!) I think a `scale="l"` Checkbox with a 24px hit area should work out fine and not make the target too much larger than the checkbox itself.

**Comment by SkyeSeitz** [comment]
It looks like 24px would meet Level AA and 44px would meet Level AAA. From a design stand point, the 30px (somewhat meeting in the middle of these requirements) doesn't seem to hurt on a 16px checkbox and adds a bit more accessibility. @geospatialem please correct me if I am wrong and 24px is your recommendation from an accessibility stand point.

**Comment by geospatialem** [comment]
> It looks like 24px would meet Level AA and 44px would meet Level AAA. From a design stand point, the 30px (somewhat meeting in the middle of these requirements) doesn't seem to hurt on a 16px checkbox and adds a bit more accessibility. @geospatialem please correct me if I am wrong and 24px is your recommendation from an accessibility stand point.

@SkyeSeitz Correct - if we go beyond the 24px we're doing great, but wouldn't be a requirement beyond the success criteria (if there were other factors at play). 👍🏻 

The AA level is what we'd aim for to fulfill the success criteria, and while AAA would be great, it's a nice to have.

**Comment by macandcheese** [comment]
Related #7223

**Comment by jcfranco** [comment]
Is this issue still valid? The now deprecated `selectable` renders a checkbox with a 24x24 hit area (AA ✅) and the new `card-group` selection modes render checkboxes with a 32x32 hit area ([example](https://codepen.io/jcfranco/pen/pomzaLP?editors=100)).

**Comment by geospatialem** [comment]
> Is this issue still valid? The now deprecated `selectable` renders a checkbox with a 24x24 hit area (AA ✅) and the new `card-group` selection modes render checkboxes with a 32x32 hit area ([example](https://codepen.io/jcfranco/pen/pomzaLP?editors=100)).

Good catch, @jcfranco! 🐟 🎣 

This is no longer valid, with the updated hitbox area of 24x24.

And below is what JAWS currently hears in context with the above example:

```
Coastal Links check box not checked 
To check press Spacebar.
Windy Plains check box checked 
To clear checkmark press Spacebar.

```

While there is no visual label present, as the original request mentions, we're fulfilling the a11y portion of the need, so will be **closing out the above**.
